gnosis/safe-react
# fix: transaction remain pending

## What it solves
Resolves #3440

## How this PR fixes it
Transactions now return the `transactionHash` from the `receipt`. After 50 blocks have been mined without successfuly execution, it throws triggering the `onError` where the pending status is removed.

`onComplete` (for on-chain) transactions now occurs upon receiving the transaction hash as we save the `txHash` for the pending transaction. (Immediately executing transactions have no `txId` and need to be proposed in order to save `{ [txId]: txHash }` in the pending store).

## How to test it
1 Create a transaction (with low gas) and then cancel it via a (high gas) cancellation in MM.
2 After 15-30 mins, observe a notification about unsuccessful pend and a console error about the transaction not being mined within 50 blocks.

## Screenshots
![image](https://user-images.githubusercontent.com/20442784/154460413-20cfa470-4ba7-4309-abfa-2d358ac9647c.png)

![image](https://user-images.githubusercontent.com/20442784/154290949-a79eaee5-a9cf-4b84-8376-073b4d398861.png)
